### PR TITLE
Build `.conda` artifacts by default

### PR DESCRIPTION
**Pull request by jaimergp**
### Description

Closes https://github.com/conda/conda-build/issues/5183

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](https://github.com/conda/conda-build/blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [x] Add / update necessary tests?
- [x] Add / update outdated documentation?

### CODE REVIEW

**Comment by jakirkham** [comment]
Thanks Jaime! 🙏

**Comment by beeankha** [comment]
It looks like there is a similar/related issue that was filed here previously: https://github.com/conda/conda-build/issues/4890

There's [currently an open PR](https://github.com/conda/conda-build/pull/5209) that addresses this issue by creating a new command line argument `--package-format` and may potentially be the option we want to go with to include in the November 2024 conda-build release. @jaimergp @jakirkham please let me know your thoughts on this.

**Comment by jakirkham** [comment]
Thanks Bianca! 🙏

Think these are different changes

This change is making the default package type `.conda` (instead of `.tar.bz2`) as requested in issue: https://github.com/conda/conda-build/issues/5183

That change is allowing users to specify the package type via CLI

These are both good changes. They just serve different purposes. Ideally we get them both in

With this change, we are thinking of having a deprecation (or future) warning in 24.11 to note the default would change in 25.1. Then include this change after the 24.11 branch (or tag)

**Comment by jaimergp** [comment]
Yea maybe https://github.com/conda/conda-build/pull/5209 should go in first (that refactor is useful here too). It's also a good opportunity to add a warning there about the future change in e.g. 25.1. In other words:

For 24.11:

- Review and merge https://github.com/conda/conda-build/pull/5209
- A new PR adds a warning about the change of the default in 25.1

For 25.1:
- Rebase this PR and flip the default. Some tests need adjustment.

**Comment by jezdez** [comment]
> Yea maybe #5209 should go in first (that refactor is useful here too). It's also a good opportunity to add a warning there about the future change in e.g. 25.1. In other words:
> 
> For 24.11:
> 
> * Review and merge [Issue/4890 #5209](https://github.com/conda/conda-build/pull/5209)
> * A new PR adds a warning about the change of the default in 25.1
> 
> For 25.1:
> 
> * Rebase this PR and flip the default. Some tests need adjustment.

I agree with that timeline, just reviewed #5209

**Review by beckermr — APPROVED** [review]
LGTM! I wonder if in the future we need to add to the test suite to cover both formats or possibly move all of the tests to `.conda` and remove support to build the old format all together.

File: conda_build/config.py [hunk 1]
```diff
@@ -101,7 +101,7 @@ def ext(self):
 no_rewrite_stdout_env_default = "false"
 ignore_verify_codes_default = []
 exit_on_verify_error_default = False
-conda_pkg_format_default = CondaPkgFormat.V1
+conda_pkg_format_default = CondaPkgFormat.V2
```
Comment (jaimergp) [inline conda_build/config.py:104]: This is the main change. Everything else is to fix assumptions about `tar.bz2` being always default.

**Pull request merged**